File: OmaProjekti/NumeronArvaus.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main () {

    //muuttumattomat arvot
    const int MIN = 1;
    const int MAX = 100;

    int vastaus = 0;
    int arvaus = 0;
    int arvauksia = 0;
    

    //hakee kellonajasta siemenluvun
    srand(time(NULL)); 
    //generoi satunnaisluvun MIN ja MAX lukujen väliltä
    vastaus = (rand() % MAX) + MIN;

    //ilmoitus ja ohje pelaajalle
    printf("\nPelataan peli!");
    printf("\nArvaa luku väliltä 1-100\n");

    //pyydetään arvaus ja ilmoitetaan onko luku pienempi vai suurempi kuin pelaajan arvaus.
    do {
        printf("\nSyötä arvauksesi: ");
        scanf("%d", &arvaus);

        if(arvaus < vastaus)
        {
            printf("Luku on suurempi. Arvaa uudelleen\n");    
        }
        else if(arvaus > vastaus)
        {
            printf("Luku on pienempi. Arvaa uudelleen\n");
        }
        else
        {
            printf("Oikein!\n");
        }
        //laskee pelaajan arvaukset
        arvauksia++;

    }while(arvaus != vastaus); //looppi pyörii kunnes arvaus on sama kuin vastaus


    printf("Vastaus: %d\n", vastaus);
    printf("Arvauksia: %d\n", arvauksia);
    printf("-------------------\n");

    return 0;

}

File: OmaProjekti/Numeronarvaus.py
import random

def main():

    #muuttumattomat arvot
    MIN = 1
    MAX = 100

    vastaus = 0
    arvaus = 0
    arvauksia = 0

    #hakee kellonajasta siemenluvun
    random.seed()
    #generoi satunnaisluvun MIN ja MAX lukujen väliltä
    vastaus = random.randint(MIN, MAX)

    #ilmoitus ja ohje pelaajalle
    print("\nPelataan peli!")
    print("Arvaa luku väliltä 1-100\n")

    #pyydetään arvaus ja ilmoitetaan onko luku pienempi vai suurempi kuin pelaajan arvaus.
    while True:
        arvaus = int(input("\nSyötä arvauksesi: "))

        if arvaus < vastaus:
            print("Luku on suurempi. Arvaa uudelleen\n")

        elif arvaus > vastaus:
            print("Luku on pienempi. Arvaa uudelleen\n")

        else:
            print("Oikein!\n")
            break

        # laskee pelaajan arvaukset
        arvauksia += 1

    print("Vastaus:", vastaus)
    print("Arvauksia:", arvauksia)
    print("-------------------")

#kutsuu main-funktiota    
main()
